topology/probes/hardware/hardware.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2020 Sylvain Afchain
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hardware
+
+import (
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/host"
+
+	"github.com/skydive-project/skydive/graffiti/graph"
+	"github.com/skydive-project/skydive/probe"
+	"github.com/skydive-project/skydive/topology/probes"
+)
+
+type Probe struct {
+	Ctx probes.Context
+}
+
+// Start the probe
+func (p *Probe) Start() error {
+	p.Ctx.Graph.Lock()
+	defer p.Ctx.Graph.Unlock()
+
+	tr := p.Ctx.Graph.StartMetadataTransaction(p.Ctx.RootNode)
+	defer tr.Commit()
+
+	if instanceID, err := getInstanceID(); err != nil {
+		p.Ctx.Logger.Error(err)
```

Comment:
I'm not sure we should log this as an error as the presence of the /var/lib/cloud/data/instance-id file is not the most common one. So you should probably use `Info` or even nothing if the file is missing

Reply:
I will use debug, so that we can still have the information if needed

Reply:
I finally added something to ignore "no such file" error